File: backend/src/main/java/com/sokhacare/modules/doctorverification/entity/DoctorVerification.java
package com.sokhacare.modules.doctorverification.entity;

import com.sokhacare.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Entity
@Table(name = "doctor_verifications")
public class DoctorVerification extends BaseEntity {

    @Column(name = "doctor_id", nullable = false, updatable = false)
    private UUID doctorId;

    @Column(name = "license_number", nullable = false, length = 100)
    private String licenseNumber;

    @Column(name = "medical_council", nullable = false, length = 120)
    private String medicalCouncil;

    @Column(name = "document_url", nullable = false, length = 500)
    private String documentUrl;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 32)
    @Builder.Default
    private DoctorVerificationStatus status = DoctorVerificationStatus.PENDING;

    @Column(name = "submitted_at", nullable = false, updatable = false)
    private Instant submittedAt;

    @Column(name = "reviewed_at")
    private Instant reviewedAt;

    @Column(name = "reviewed_by")
    private UUID reviewedBy;

    @Column(name = "remarks", length = 1000)
    private String remarks;
}

File: src/types/doctorVerification.ts
export type DoctorVerificationStatus = 'PENDING' | 'UNDER_REVIEW' | 'APPROVED' | 'REJECTED';

export type DoctorVerification = {
  id: string;
  doctorId: string;
  licenseNumber: string;
  medicalCouncil: string;
  documentUrl: string;
  status: DoctorVerificationStatus;
  submittedAt: string;
  reviewedAt: string | null;
  reviewedBy: string | null;
  remarks: string | null;
};

export type DoctorVerificationStatusResponse = {
  doctorId: string;
  doctorProfileExists: boolean;
  canSubmit: boolean;
  latestVerification: DoctorVerification | null;
};

export type DoctorVerificationSubmitValues = {
  licenseNumber: string;
  medicalCouncil: string;
  documentUrl: string;
};

export type DoctorVerificationReviewValues = {
  remarks: string;
};
